src/prpl.ts

```diff
@@ -26,6 +26,14 @@ export interface Config {
   // Defaults to `max-age=60`.
   cacheControl?: string;
 
+  // A custom error-handling logic function. The function must takes two 
+  // arguments, request and response, and must return a function with
+  // the argument error.
```

Comment:
Why does it need to be a function that returns a function? Couldn't you just pass `request`, `response`, and `error` as 3 params to this function directly?

Reply:
The error listener of the send stream will be called with only one argument, the HTTP error (see [L269-L272](https://github.com/pillarjs/send/blob/3daa901cf731b86187e4449fa2c52f971e0b3dbc/index.js#L269-L272) of send). So, in order for the request and the response to be accessible inside the error function, I wrote it as you have seen it.

But I think we could wrote it as a simple function with three parameters if you want, by remplacing [L183](https://github.com/ranb2002/prpl-server-node/blob/5d9fe8607c8b3c7d0c126369b56f6efcd22231ac/src/prpl.ts#L183) with 
```typescript
stream = stream.on('error', (error: httpErrors.HttpError) => config.error(request, response, error));
```
and [L99](https://github.com/ranb2002/prpl-server-node/blob/5d9fe8607c8b3c7d0c126369b56f6efcd22231ac/src/prpl.ts#L99) - [L126](https://github.com/ranb2002/prpl-server-node/blob/5d9fe8607c8b3c7d0c126369b56f6efcd22231ac/src/prpl.ts#L126) with
```typescript
config.error(request, response, httpErrors(403));
config.error(request, response, httpErrors(500));
```

I will try it this week.

Reply:
It's working, with the ! non-null assertion operator. :smiley:
```typescript
    // Set a custom error-handling function
    if (config && config.error) {
      const errorFunction = (error:  httpErrors.HttpError) => config.error!(request, response, error)
      stream = stream.on('error', errorFunction);
    }
```